account_payment_order/models/account_payment_order.py

```diff
@@ -462,7 +462,7 @@ def _prepare_move_line_offsetting_account(
             vals.update({"date_maturity": bank_lines[0].date})
 
         if self.payment_mode_id.offsetting_account == "bank_account":
-            account_id = self.journal_id.default_account_id.id
+            account_id = self.journal_id.payment_debit_account_id.id
```

Comment:
you should make a difference between inbound and outbond : payment_debit_account_id and payment_credit_account_id

Reply:
Agree. @max3903 Could you update accordingly?

Reply:
@alexis-via That's a behaviour you partly removed here: https://github.com/OCA/bank-payment/pull/822/files#diff-f55b2fce8819072c65c79eaaf78564a96202ccca3de7a8be6f99c1c6862cb325L430

Reply:
Did it there : https://github.com/ursais/bank-payment/pull/9

Reply:
Ok, I confirm this has already been solved in latest version. So, no need this change